app/models/provided/exchange_rate.rb

```diff
@@ -0,0 +1,22 @@
+class Provided::ExchangeRate
+  include Providable, Retryable
+
+  def initialize
+    @provider = exchange_rates_provider
+  end
+
+  def fetch(from:, to:, date:)
+    retrying Provider::Base.known_transient_errors do
```

Comment:
Haven't given any thought to how rate limiting comes into play. I think `Provided::<Concept>` classes can be aware of how many requests have been performed via `Providable` and modify their actions accordingly.